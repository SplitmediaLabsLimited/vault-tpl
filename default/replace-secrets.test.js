const parse = require('./parse');
const { replaceSecrets } = require('./replace-secrets');

test('multiple and duplicate secrets, should deduplicate', () => {
  const content = `
  secret: (( vault "version-test:secret@2" ))
  secret: (( vault "version-test:secret@2" ))
  secret: (( vault "version-test:secret@2" ))
  secret2: (( vault "version-test:secret2@3" ))
  secret3: "raw"
  `;

  const matches = parse(content);
  const secrets = matches.map((match, index) => ({
    ...match,
    secret: `secret values goes here ${index}`,
  }));

  const replaced = replaceSecrets(content, secrets);

  expect(replaced).toMatchSnapshot();
});

test('no secrets found', () => {
  const content = `
  secret: (( vault "version-test:secret@2" ))
  secret: (( vault "version-test:secret@2" ))
  secret: (( vault "version-test:secret@2" ))
  secret2: (( vault "version-test:secret2@3" ))
  secret3: "raw"
  `;

  const matches = parse(content);
  const secrets = matches.map((match, index) => ({
    ...match,
    secret: undefined,
  }));

  const replaced = replaceSecrets(content, secrets);

  expect(replaced).toMatchSnapshot();
});
