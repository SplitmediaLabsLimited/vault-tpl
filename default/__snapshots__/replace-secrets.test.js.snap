// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`multiple and duplicate secrets, should deduplicate 1`] = `
"
  secret: secret values goes here 0
  secret: secret values goes here 0
  secret: secret values goes here 0
  secret2: secret values goes here 1
  secret3: \\"raw\\"
  "
`;

exports[`no secrets found 1`] = `
"
  secret: 
  secret: 
  secret: 
  secret2: 
  secret3: \\"raw\\"
  "
`;
